### PR TITLE
workaround odd shift behaviour

### LINKED ISSUE #397: odd shift key behavior
I have observed that if I press both shift keys together and then release one, I get lower-case characters even though one shift key is still being held.  "xev" confirms I get 2 KeyRelease events when a single shift key is released.  I've noted identical behavior with both Xvnc and xorgxrdp sessions.  By comparison, if I bypass Xrdp and use VNC directly, I don't have this problem.
I'm really not sure where to start looking for this problem.  Is it a key mapping problem or something else?  Any pointers on where to start looking would be appreciated.

TIA

### PR DESCRIPTION
**Pull request by okhowang**
Fix #397
Workaround by ignore odd shift message

### CODE REVIEW

**Comment by okhowang** [comment]
@metalefty is there any other problem to merge?

**Comment by metalefty** [comment]
Overall, LGTM but I haven't tested yet. 
@jsorg71 what do you think?

File: xrdp/xrdp_wm.c [hunk 1]
```diff
@@ -1532,6 +1532,12 @@ xrdp_wm_key(struct xrdp_wm *self, int device_flags, int scan_code)
         return 0;
     }
 
+    // walkaround odd shift behavior
```
Comment (metalefty) [inline xrdp/xrdp_wm.c:1535]: BTW, you mean "workaround".

Comment (okhowang) [reply xrdp/xrdp_wm.c]: Fixed, あの時、頭がおかしくなったかも

**Comment by metalefty** [comment]
@okhowang 
Hi, thanks for your patience. I've just tested your pull request however the fix is incomplete.
Can you have a look again?

## OK
1. Press left shift (keep pressed)
2. Press "A" multiple times
    * got "AAAAAAAAA" input
3. Press right shift and release right shift
4. Press "A" key multiple times  (left shift is still being pressed)
    * got "AAAAAAAAA" input (as expected)

## NG (swap right and left shift)
1. Press right shift (keep pressed)
2. Press "A" multiple times
    * got "AAAAAAAAA" input
3. Press left shift and release left shift
4. Press "A" key multiple times  (right shift is still being pressed)
    * "a" becomes lowercase (uppercase expected)
    * got "AAAAAAAAAaaaaaaaaa"

**Comment by metalefty** [comment]
## Left shift first

1. Press left shift
2. Press right shift
3. Release right shift

Got this (as expected):
```
KeyPress event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x2200002, time 3100008, (58,61), root:(929,516),
    state 0x0, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyPress event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x2200002, time 3101562, (58,61), root:(929,516),
    state 0x1, keycode 62 (keysym 0xffe2, Shift_R), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x2200002, time 3101697, (58,61), root:(929,516),
    state 0x1, keycode 62 (keysym 0xffe2, Shift_R), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```
## Right shift first

1. Press right shift
2. Press left shift
3. Release left shift

```
 KeyPress event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x0, time 2931544, (-151,273), root:(720,728),
    state 0x0, keycode 62 (keysym 0xffe2, Shift_R), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyPress event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x0, time 2933253, (-151,273), root:(720,728),
    state 0x1, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 37, synthetic NO, window 0x2200001,
    root 0x25e, subw 0x0, time 2933374, (-151,273), root:(720,728),
    state 0x1, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 37, synthetic NO, window 0x2200001,              <- this shouldn't happen
    root 0x25e, subw 0x0, time 2933374, (-151,273), root:(720,728),
    state 0x1, keycode 62 (keysym 0xffe2, Shift_R), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

**Comment by okhowang** [comment]
see https://github.com/neutrinolabs/xrdp/issues/397#issuecomment-599477359
the second situation is remain odd in microsoft implementation.
so I just keep same behaviour with it.
DO we need fix the second?

**Comment by metalefty** [comment]
Okay, understood.

**Pull request merged**